src/chunked/encoder.rs

```diff
@@ -72,19 +72,37 @@ impl ChunkedEncoder {
         buf: &mut [u8],
     ) -> Poll<io::Result<usize>> {
         self.bytes_written = 0;
+        let res = self.exec(res, cx, buf);
+        log::trace!("ChunkedEncoder {} bytes written", self.bytes_written);
+        res
+    }
+
+    /// Execute the right method for the current state.
+    fn exec(
+        &mut self,
+        res: &mut Response,
+        cx: &mut Context<'_>,
+        buf: &mut [u8],
+    ) -> Poll<io::Result<usize>> {
         match self.state {
-            State::Start => self.init(res, cx, buf),
+            State::Start => self.set_state(State::EncodeChunks, res, cx, buf),
             State::EncodeChunks => self.encode_chunks(res, cx, buf),
             State::EndOfChunks => self.encode_chunks_eos(res, cx, buf),
-            State::ReceiveTrailers => self.encode_trailers(res, cx, buf),
+            State::ReceiveTrailers => self.receive_trailers(res, cx, buf),
```

Comment:
Actually found a bug here -- inconsequential, but rustc is now better able to infer whether code is used.